### ISSUE TITLE
Question about RSI calculation

### ISSUE THREAD

**Issue by Shawn8901** [DeviaVir/zenbot#1056]
### System information
- **Have I written custom code (as opposed to using zenbot vanilla)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Debian 9.3
- **Zenbot version**:  4.05
- **NodeJS version**: 9.3.0
- **Python version**: ??
- **Exact command to reproduce**: any strategy, since RSI is displayed all the time

### Describe the problem
Currently i am working on a custom strategy and i am checking the calculated indicator values with some charts.
Here the point i am confused about:
 
As i understand the RSI calculation current it should sum all candle wins and candle losses (on close value in case of zenbot). 
And then does its magic calculation (not important here).
If i am looking at the following code

        s.lookback.slice(0, length).forEach(function (period) {
          if (last_close) {
            if (period.close > last_close) {
              gain_sum += period.close - last_close
            }
            else {
              loss_sum += last_close - period.close
            }
          }
          last_close = period.close
        })

Lets assume that we have RSI_Length of 3 and 4 Lookback data of [6 (youngest) , 7 , 5, 4 (oldest, but not in RSI_Length)]
It will start with the youngest lookback and will use the following numbers (6 -> 7 -> 5) for calculation.
So the calculation will be the following, if i am understanding it right:
1. iteration last_close = NULL --> use 6
2. iteration 7 > 6 so gain_sum = 1
3. iteration 5 < 7 so loss_sum = 2

On my option the calculation has to be reversed, because actually moving from  lookback[2] to lookback[1] is a gain and not a loss!
Right values should be:
gain_sum =  2  (candle 5 --> 7) 
loss_sum = 1 (candle 7 --> 6)

So i think there is an reverse() missing before iteration?
 
Can some one with more experience please re-check this?

**Comment by dshetyo** [comment]
Why is this closed ? It's a bug and not enhancements

**Comment by Shawn8901** [comment]
It was a question and no one answered or showed any interest so i closed my question.

**Comment by DeviaVir** [comment]
@Shawn8901 I think we can leave it open for now.

**Comment by aquamanjl** [comment]
@Shawn8901 I have created a PR https://github.com/DeviaVir/zenbot/pull/1168 to fix an issue with RSI calculation that resulted in NaN values. As part of the same PR I also added test cases to validate the RSI lib and it appears to be working as expected. I took the data from stockcharts and the results do match.

I must say I am not a fan of rounding off the numbers but I didn't want to break anything so left it as is. Can live with it for now.
